ServiceTier/Services/TaskService.cs

```diff
@@ -0,0 +1,230 @@
+            {
+                throw new TaskNotFoundException();
+            }
+        }
+
+        public IQueryable<TaskDTO> GetAllTasks()
+        {
+            var listOfTasksDTO = new List<TaskDTO>();
+            var tasks = unitOfWork.Tasks.GetAll();
+
+            if (tasks == null)
+            {
+                throw new Exception("There are no tasks!");
+            }
+
+            foreach (var task in tasks)
+            {
+                listOfTasksDTO.Add(task.ToTaskDTO());
+            }
+
+            return listOfTasksDTO.AsQueryable();
+        }
+
+        public IQueryable<TaskDTO> GetAllTasksByUserId(int userId)
+        {
+            var listOfTasksDTO = new List<TaskDTO>();
+            var tasks = unitOfWork.Tasks.GetAll();
+
+            if (tasks == null)
+            {
+                throw new Exception("There are no tasks!");
+            }
+
+            foreach (var task in tasks)
+            {
+                if (task.UserId == userId)
+                {
+                    listOfTasksDTO.Add(task.ToTaskDTO());
+                }
+            }
+
+            return listOfTasksDTO.AsQueryable();
+        }
+
+        public TaskDTO GetTaskById(int taskId)
+        {
+            Tasks task = unitOfWork.Tasks.GetByID(taskId);
+            if (task == null)
+            {
+                throw new TaskNotFoundException();
+            }
+            return task.ToTaskDTO();
+        }
+
+        public void UpdateTaskById(TaskDTO taskDTO)
+        {
+            var taskId = taskDTO.Id;
+            Tasks task = unitOfWork.Tasks.GetByID(taskId);
+
+            if (task == null)
+            {
+                throw new TaskNotFoundException();
+            }
+
+            task.Name = taskDTO.Name;
+            task.Description = taskDTO.Description;
+            task.Goal = taskDTO.Goal;
+            task.IsActive = taskDTO.IsActive;
+            task.IsRunning = taskDTO.IsRunning;
+
+            unitOfWork.Tasks.Update(task);
+            unitOfWork.Save();
+        }
+
+        public void UpdateTaskStatus(int taskId, bool newStatus)
+        {
+            Tasks task = unitOfWork.Tasks.GetByID(taskId);
+            if (task == null)
+            {
+                throw new TaskNotFoundException();
+            }
+            task.IsActive = newStatus;           
+            
+            unitOfWork.Tasks.Update(task);
+            unitOfWork.Save();
+        }
+
+        public void ResetTaskById(int taskId)
+        {
+            var task = unitOfWork.Tasks.GetByID(taskId);
+            if (task == null)
+            {
+                throw new TaskNotFoundException();
+            }
+            task.ElapsedTime = TimeSpan.Zero;
+            task.Goal = null;
+            task.LastStartTime = task.LastStartTime.Date;
+            task.IsRunning = false;
+
+            unitOfWork.Tasks.Update(task);
+            unitOfWork.Save();
+        }
+
+        public void StartTimer(int taskId)
```

Comment:
"timer" is a bit confusing because we have timers in presets. "StartTask" sounds good enough for me